public/app/features/alerting/unified/components/rules/EditRuleGroupModal.tsx

```diff
@@ -198,31 +211,36 @@ export function EditCloudGroupModal(props: ModalProps): React.ReactElement {
 
   const nameSpaceLabel = isGrafanaManagedGroup ? 'Folder' : 'Namespace';
 
-  // close modal if successfully saved
-  useEffect(() => {
-    if (dispatched && !loading && !error) {
-      onClose(true);
-    }
-  }, [dispatched, loading, onClose, error]);
+  const onSubmit = async (values: FormValues) => {
+    const ruleGroupIdentifier: RuleGroupIdentifier = {
+      dataSourceName: rulesSourceName,
+      groupName: group.name,
+      namespaceName: isGrafanaManagedGroup ? folderUid! : namespace.name,
+    };
 
-  useCleanup((state) => (state.unifiedAlerting.updateLotexNamespaceAndGroup = initialAsyncRequestState));
-  const onSubmit = (values: FormValues) => {
     // make sure that when dealing with a nested folder for Grafana managed rules we encode the folder properly
-    const newNamespaceName = isGrafanaManagedGroup
+    const updatedNamespaceName = isGrafanaManagedGroup
       ? encodeGrafanaNamespace(values.namespaceName, nestedFolderParents)
       : values.namespaceName;
+    const updatedGroupName = values.groupName;
+    const updatedInterval = values.groupInterval;
 
-    dispatch(
-      updateLotexNamespaceAndGroupAction({
-        rulesSourceName: rulesSourceName,
-        groupName: group.name,
-        newGroupName: values.groupName,
-        namespaceName: namespace.name,
-        newNamespaceName: newNamespaceName,
-        groupInterval: values.groupInterval || undefined,
-        folderUid,
-      })
-    );
+    // GMA alert rules cannot be moved to another folder, we currently do not support it but it should be doable (with caveats).
+    const shouldMove = isGrafanaManagedGroup ? false : updatedNamespaceName !== ruleGroupIdentifier.namespaceName;
+    const shouldRename = updatedGroupName !== ruleGroupIdentifier.groupName;
+
+    if (shouldMove) {
+      await moveRuleGroup(ruleGroupIdentifier, updatedNamespaceName, updatedGroupName, updatedInterval);
+      onClose(true);
+    } else if (shouldRename) {
+      await renameRuleGroup(ruleGroupIdentifier, updatedGroupName, updatedInterval);
+      onClose(true);
+    } else {
+      await updateRuleGroup(ruleGroupIdentifier, updatedInterval);
+      onClose(true);
+    }
+
+    await dispatch(fetchRulerRulesAction({ rulesSourceName }));
```

Comment:
Just thinking, could we instead put this dispatch into the `onCacheEntryAdded` of our mutations, as a pseudo-tag-invalidation? 🤔

Reply:
I think we should not, as it may be that in some cases we don't want to fetch all the rules? .... And, eventually this dispatch will disappear?? as we will move getting rules it to RTKQ?

Reply:
I'm with @soniaAguilarPeiron on this one – feels like we should defer this and just keep pushing forward to move everything to RTKQ with proper tag invalidation.

Reply:
Yep, to be clear, not necessarily something for here, I just realised we could probably solve it in that was as well :+1: